DelayedMessages/explainer.md

```diff
@@ -0,0 +1,757 @@
+
+# Security and Privacy Considerations
+
+This API is designed to provide developers with insights into the performance of their own applications and does not introduce new cross-origin information leakage.
+
+  * **Same-Origin Data:** `PerformanceDelayMessageTiming` entries are only exposed to the same origin that initiated and received the messages. Information about message timings between `A.com` and an iframe from `B.com` would only be available to `A.com` if it can observe the receiver in `B.com` (which generally requires `B.com` to cooperate) or vice-versa, subject to standard cross-origin policies. The `scripts` property, which details blocking tasks, also only contains information about same-origin scripts.
+  * **No New Cross-Origin Information:** The API does not expose any new information about cross-origin interactions that isn't already observable through other means (e.g., whether a message was sent or received). Details of cross-origin execution contexts or script contents are not revealed.
+  * **Fingerprinting:** The risk of this API being used for user fingerprinting is considered low.
+      * The information provided is about the performance characteristics of the web application itself, not the user's underlying system hardware beyond what influences JavaScript execution speed.
+      * The `durationThreshold` for observing entries limits the granularity and volume of data reported, making it less suitable for creating a detailed fingerprint.
+      * While timing variations could exist, they are more likely attributable to the application's state and workload rather than unique user environment characteristics.
+
+# Discussion
+
+## Minimum Duration for Delayed Messages (`durationThreshold`)
+
+We propose a default `durationThreshold` of 200ms for identifying delayed messages. However, the impact of a delay can be context-dependent. For example, a 100ms delay might not cause any issues in a background worker task, but the same delay could be noticeable to users if it affects communication between an iframe and its parent window during a user interaction.
+
+To address this, the API makes `durationThreshold` configurable by the developer. This allows developers to adjust the threshold to their specific application's performance requirements and specific use cases. In addition, the `durationThreshold` also helps ensure that data is only collected for messages that are genuinely problematic, reducing unnecessary monitoring overhead.
+
+Further discussion is needed to determine the minimum allowed value for this threshold (e.g., 50ms as suggested) to ensure it remains useful without introducing significant performance costs.
+
+# References
+- [Extending Long Tasks API to Web Workers](https://github.com/joone/MSEdgeExplainers/blob/add_id_src_type/LongTasks/explainer.md)
+- https://developer.mozilla.org/en-US/docs/Web/API/PerformanceLongTaskTiming

```

Comment:
The event timing API, https://w3c.github.io/event-timing/, might be another resource to draw from since postMessage produces MessageEvent.  The API is focused on measure long input event responsible for UI hangs.

Reply:
Okay, I’ll include the Event Timing API as a reference.